freshmaker/views.py

```diff
@@ -386,6 +385,24 @@ def _validate_rebuild_request(request):
         if data.get(key) and not isinstance(data[key], str):
             return json_error(400, 'Bad Request', f'"{key}" must be a string.')
 
+    def _event_state_check(freshmaker_event_id):
+        event = models.Event.get_by_event_id(db.session, freshmaker_event_id)
+        event_state = event.state
+        if event_state in (0, 1):
+            return json_error(400, 'Bad Request', f'"{event.id}" is in Initialized or Building state. If you want to continue, use "force": True option.')
+
+    if not data.get("force", None):
+        if 'freshmaker_event_id' in data:
+            _event_state_check(data.get('freshmaker_event_id'))
+        if 'errata_id' in data:
+            events = []  # Todo
```

Comment:
@qixiang Is there any method or function which would return a list of events by errata_id.

Reply:
`search_key` in `Event` model is the errata id.

We don't need to check against `freshmaker_event_id` and `errata_id` separately since errata id is available in event, a better place to perform such a check is before:
https://github.com/redhat-exd-rebuilds/freshmaker/blob/c8807d784990b058347ca22e56d839a4718e6d8f/freshmaker/views.py#L556-L559

with something like:
```
running_events = (
    models.Event.query.filter(models.Event.search_key == str(data['errata_id']))
    .filter(models.Event.state.in_([EventState.INITIALIZED.value, EventState.BUILDING.value]))
    .all()
)
```